pkg/controller/spinnakeraccount/spinnakeraccount_controller.go

```diff
@@ -0,0 +1,148 @@
+package spinnakeraccount
+
+import (
+	"context"
+	"github.com/armory/spinnaker-operator/pkg/accounts"
+	"github.com/armory/spinnaker-operator/pkg/accounts/account"
+	"github.com/armory/spinnaker-operator/pkg/accounts/find"
+	"github.com/armory/spinnaker-operator/pkg/apis/spinnaker/v1alpha2"
+	"github.com/armory/spinnaker-operator/pkg/secrets"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	logf "sigs.k8s.io/controller-runtime/pkg/runtime/log"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+)
+
+var log = logf.Log.WithName("spinnakerservice")
+
+var SpinnakerServiceBuilder v1alpha2.SpinnakerServiceBuilderInterface
+
+// Add creates a new SpinnakerService Controller and adds it to the Manager. The Manager will set fields on the Controller
+// and Start it when the Manager is Started.
+func Add(mgr manager.Manager) error {
+	return add(mgr, newReconciler(mgr))
+}
+
+// newReconciler returns a new reconcile.Reconciler
+func newReconciler(mgr manager.Manager) reconcile.Reconciler {
+	return &ReconcileSpinnakerAccount{
+		client: mgr.GetClient(),
+		scheme: mgr.GetScheme(),
+	}
+}
+
+// add adds a new Controller to mgr with r as the reconcile.Reconciler
+func add(mgr manager.Manager, r reconcile.Reconciler) error {
+	// Create a new controller
+	c, err := controller.New("spinnakeraccount-controller", mgr, controller.Options{Reconciler: r})
+	if err != nil {
+		return err
+	}
+
+	// Watch for changes to primary resource SpinnakerService
+	err = c.Watch(&source.Kind{Type: &v1alpha2.SpinnakerAccount{}}, &handler.EnqueueRequestForObject{})
```

Comment:
Do we need to watch for `SpinnakerService` in case the acounts enabled/dynamic flag is changed?

Reply:
I don't think so. If `SpinnakerService` changes, it will be reconciled and get the CRD accounts on its own.